src/foundry/client-esm/canvas/smaa/smaa.d.mts

```diff
@@ -1 +1,74 @@
-export default class SMAAFilter extends PIXI.Filter {}
+import type { CLEAR_MODES, FilterState, FilterSystem, RenderTexture } from "pixi.js";
+
+import type { InexactPartial } from "../../../../types/utils.d.mts";
+
+declare class SMAAFilter extends PIXI.Filter {
+  /**
+   * @param config    - The config (defaults: SMAAFilter.PRESETS.DEFAULT)
+   */
+  constructor(config?: InexactPartial<SMAAFilter.SMAAFilterConfig>);
+
+  // a placeholder private method to help subclassing
+  #smaaFilter: true;
+
+  /**
+   * The presets.
+   */
+  static get PRESETS(): Record<string, SMAAFilter.SMAAFilterConfig>;
+
+  /** @override */
+  apply(
+    filterManager: FilterSystem,
+    input: RenderTexture,
+    output: RenderTexture,
+    clearMode: CLEAR_MODES | undefined,
+    currentState: FilterState,
+  ): void;
+}
+
+declare namespace SMAAFilter {
+  interface SMAAFilterConfig {
+    /**
+     * Specifies the threshold or sensitivity to edges. Lowering this value you will be able to detect more edges at the expense of performance.
+     * Range: [0, 0.5]. 0.1 is a reasonable value, and allows to catch most visible edges. 0.05 is a rather overkill value, that allows to catch 'em all.
+     */
+    threshold: number;
+
+    /**
+     * If there is an neighbor edge that has SMAA_LOCAL_CONTRAST_FACTOR times bigger contrast than current edge, current edge will be discarded.
+     * This allows to eliminate spurious crossing edges, and is based on the fact that, if there is too much contrast in a direction, that will
+     *   hide perceptually contrast in the other neighbors.
+     */
+    localContrastAdaptionFactor: number;
+
+    /**
+     * Specifies the maximum steps performed in the horizontal/vertical pattern searches, at each side of the pixel. In number of pixels, it's
+     *   actually the double. So the maximum line length perfectly handled by, for example 16, is 64 (by perfectly, we meant that longer lines
+     *   won't look as good, but still antialiased. Range: [0, 112].
+     */
+    maxSearchSteps: number;
+
+    /**
+     * Specifies the maximum steps performed in the diagonal pattern searches, at each side of the pixel. In this case we jump one pixel at time,
+     *   instead of two. Range: [0, 20].
+     */
+    maxSearchStepsDiag: number;
```

Comment:
So these leading spaces aren't necessary. I'm pretty sure they get collapsed anyways. 

In general we leave the comments as exactly how they are in Foundry. This makes it easier to diff comments against Foundry.

Reply:
should put them inline next to the properties?  these were a typedef before:

```
/**
 * @typedef {object} SMAAFilterConfig
 * @property {number} threshold                    Specifies the threshold or sensitivity to edges. Lowering this value you will be able to detect more edges at the expense of performance. Range: [0, 0.5]. 0.1 is a reasonable value, and allows to catch most visible edges. 0.05 is a rather overkill value, that allows to catch 'em all.
 * @property {number} localContrastAdaptionFactor  If there is an neighbor edge that has SMAA_LOCAL_CONTRAST_FACTOR times bigger contrast than current edge, current edge will be discarded.
 *                                                 This allows to eliminate spurious crossing edges, and is based on the fact that, if there is too much contrast in a direction, that will hide perceptually contrast in the other neighbors.
 * @property {number} maxSearchSteps               Specifies the maximum steps performed in the horizontal/vertical pattern searches, at each side of the pixel. In number of pixels, it's actually the double. So the maximum line length perfectly handled by, for example 16, is 64 (by perfectly, we meant that longer lines won't look as good, but still antialiased. Range: [0, 112].
 * @property {number} maxSearchStepsDiag           Specifies the maximum steps performed in the diagonal pattern searches, at each side of the pixel. In this case we jump one pixel at time, instead of two. Range: [0, 20].
 * @property {number} cornerRounding               Specifies how much sharp corners will be rounded. Range: [0, 100].
 * @property {boolean} disableDiagDetection        Is diagonal detection disabled?
 * @property {boolean} disableCornerDetection      Is corner detection disabled?
 */

```

Reply:
Yeah, inline against the properties but otherwise the same. I don't think it works otherwise. This makes it so that you can blindly copy and paste over once you've generated with a `.d.ts` file.

Reply:
hmmm... I tried it, but prettier pulled the 2nd line of the 2-line comment over to the left.  See what you think:
```
 interface SMAAFilterConfig {
    threshold: number; // Specifies the threshold or sensitivity to edges. Lowering this value you will be able to detect more edges at the expense of performance. Range: [0, 0.5]. 0.1 is a reasonable value, and allows to catch most visible edges. 0.05 is a rather overkill value, that allows to catch 'em all.
    localContrastAdaptionFactor: number; // If there is an neighbor edge that has SMAA_LOCAL_CONTRAST_FACTOR times bigger contrast than current edge, current edge will be discarded.
    //This allows to eliminate spurious crossing edges, and is based on the fact that, if there is too much contrast in a direction, that will hide perceptually contrast in the other neighbors.
    maxSearchSteps: number; // Specifies the maximum steps performed in the horizontal/vertical pattern searches, at each side of the pixel. In number of pixels, it's actually the double. So the maximum line length perfectly handled by, for example 16, is 64 (by perfectly, we meant that longer lines won't look as good, but still antialiased. Range: [0, 112].
    maxSearchStepsDiag: number; // Specifies the maximum steps performed in the diagonal pattern searches, at each side of the pixel. In this case we jump one pixel at time, instead of two. Range: [0, 20].
    cornerRounding: number; // Specifies how much sharp corners will be rounded. Range: [0, 100].
    disableDiagDetection: boolean; // Is diagonal detection disabled?
    disableCornerDetection: boolean; // Is corner detection disabled?
  }
  ```

Reply:
you want it to look like
```ts
interface SMAAFilterConfig {
  /** Specifies the threshold or sensitivity to edges. Lowering this value you will be able to detect more edges at the expense of performance. Range: [0, 0.5]. 0.1 is a reasonable value, and allows to catch most visible edges. 0.05 is a rather overkill value, that allows to catch 'em all. */
  threshold: number;
}
```
using that /** */ syntax. Feel free to adjust line breaks - because we're reformatting TypeDef to interface, what makes sense for line breaks is going to change, and the actual type hints don't care.

Reply:
That's actually what I had in the prior revision.  I may have misunderstood, but I think it's different from what @LukeAbby was asking for above. :)

Reply:
Ah okay if Prettier is reformatting it then we have no choice. Please do leave it as `/**` as otherwise they're not documentation comments. though the leading spaces you had on the second lines are still unnecessary.

I might try to get Prettier to stop formatting comments; I want these to be very easy to diff with Foundry source.

Reply:
Check out this version.  Changed back to JSDoc comments.  Anything that was one line I left as one line.  The one that was multiple lines I split where they did.  Not sure how easy it will be to diff since the structure is different (the comments in the source are on the same line as the properties) but this is as close as I think we can get it while still using JSDoc comments.